File: leetcode-submissions/53_Maximum_Subarray.ts
function maxSubArray(nums: number[]): number {
    let maxSum:number=Number.MIN_SAFE_INTEGER;
    let tempSum:number=0;
    for(let i:number=0;i<nums.length;i++){
        tempSum+=nums[i];
        if(maxSum<tempSum){
            maxSum=tempSum;
        }
        if(tempSum<0){
            tempSum=0;
        }
    }
    return  maxSum;
};

File: leetcode-submissions/53_Maximum_Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxele=INT_MIN;
        int sumtemp=0;
        for(int i=0;i<nums.size();i++){
            sumtemp+=nums[i];
            if(maxele<sumtemp){
                maxele=sumtemp;
            }
            if(sumtemp<0){
                sumtemp=0;
            }
        }
        return maxele;
    }
};